tests/pystone.py

```diff
@@ -23,245 +23,239 @@
-                    Array2Par[IntLoc][IntIndex] = IntLoc
-                Array2Par[IntLoc][IntLoc-1] = Array2Par[IntLoc][IntLoc-1] + 1
-                Array2Par[IntLoc+20][IntLoc] = Array1Par[IntLoc]
-                IntGlob = 5
-
-            def Func1(CharPar1, CharPar2):
-                CharLoc1 = CharPar1
-                CharLoc2 = CharLoc1
-                if CharLoc2 != CharPar2:
-                    return Ident1
-                else:
-                    return Ident2
-
-            def Func2(StrParI1, StrParI2):
-                IntLoc = 1
-                while IntLoc <= 1:
-                    if Func1(StrParI1[IntLoc], StrParI2[IntLoc+1]) == Ident1:
-                        CharLoc = 'A'
-                        IntLoc = IntLoc + 1
-                if CharLoc >= 'W' and CharLoc <= 'Z':
-                    IntLoc = 7
-                if CharLoc == 'X':
-                    return TRUE
-                else:
-                    if StrParI1 > StrParI2:
-                        IntLoc = IntLoc + 7
-                        return TRUE
-                    else:
-                        return FALSE
-
-            def Func3(EnumParIn):
-                EnumLoc = EnumParIn
-                if EnumLoc == Ident3: return TRUE
-                return FALSE
-
-            if __name__ == '__main__':
-                import sys
-                def error(msg):
-                    print(msg, end=' ', file=sys.stderr)
-                    print("usage: %s [number_of_loops]" % sys.argv[0], file=sys.stderr)
-                    sys.exit(100)
-                nargs = len(sys.argv) - 1
-                if nargs > 1:
-                    error("%d arguments are too many;" % nargs)
-                elif nargs == 1:
-                    try: loops = int(sys.argv[1])
-                    except ValueError:
-                        error("Invalid argument %r;" % sys.argv[1])
-                else:
-                    loops = LOOPS
-                main(loops)
-        """))
-        print(out)
+LOOPS = 50000
+
+from time import clock
+
+__version__ = "1.1"
+
+[Ident1, Ident2, Ident3, Ident4, Ident5] = range(1, 6)
+
+class Record:
+
+    def __init__(self, PtrComp = None, Discr = 0, EnumComp = 0,
+                       IntComp = 0, StringComp = 0):
+        self.PtrComp = PtrComp
+        self.Discr = Discr
+        self.EnumComp = EnumComp
+        self.IntComp = IntComp
+        self.StringComp = StringComp
+
+    def copy(self):
+        return Record(self.PtrComp, self.Discr, self.EnumComp,
+                      self.IntComp, self.StringComp)
+
+TRUE = 1
+FALSE = 0
+
+def main(loops=LOOPS):
+    benchtime, stones = pystones(loops)
+    print("Pystone(%s) time for %d passes = %g" % \
+          (__version__, loops, benchtime))
+    print("This machine benchmarks at %g pystones/second" % stones)
+
+
+def pystones(loops=LOOPS):
+    return Proc0(loops)
+
+IntGlob = 0
+BoolGlob = FALSE
+Char1Glob = '\\0'
+Char2Glob = '\\0'
```

Comment:
These null bytes have been escaped, this is the only difference from the source.